File: ConsoleApp1/Adder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    class Adder:IProcess
    {
        public Adder() { }
        public dynamic Process(dynamic A, dynamic B)
        {
            return A + B;
        }
        public int ModProcess(int A, int B, int M)
        {
            return (A + B) % M;
        }
    }
}

File: src/com/company/Adder.java
package com.company;

public class Adder implements Process {
    public Adder() {
    }

    @Override
    public double process(double a, double b) {
        return a + b;
    }

    @Override
    public int modProcess(int a, int b, int m) {
        return (a + b) % m;
    }
}
